### DISCUSSION TITLE
Cargo tarpaulin very slow

### DISCUSSION THREAD

**Discussion by Aeradriel** [xd009642/tarpaulin#1143] [Q&A]
Hi!

I'm facing an issue trying to use tarpaulin on my workspace. Everything is "working" but between the tests runs, tarpaulin seems to hang for a very l long time.

When a test file runs, it finishes rather quickly (no longer than in normal cargo test I would say) and after those lines, it hangs for what feels like an eternity.

```
test result: ok. 5 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 2.52s

Nov 22 21:19:26.861  INFO cargo_tarpaulin::statemachine::instrumented: For binary: ../../target/tarpaulin/debug/deps/log_as-1115b718754e81ff
Nov 22 21:19:26.863  INFO cargo_tarpaulin::statemachine::instrumented: Generated: log_as-1115b718754e81ff_8453474060636193777_0-55402.profraw
```

Then, after long minutes, it "starts again":

```
Nov 22 21:25:51.525  INFO cargo_tarpaulin::process_handling: running XXX/target/tarpaulin/debug/deps/results-5dbcc43495e12fe8
Nov 22 21:25:51.525  INFO cargo_tarpaulin::process_handling: Setting LLVM_PROFILE_FILE
```

I tried several ways to start tarpaulin (mac, linux & docker) but no one seems to be quicker.
Here is how I (tried to) run it:

```
cargo tarpaulin --workspace
cargo tarpaulin --skip-clean
cargo tarpaulin --target-dir ./target/tarpaulin
```

Is there anything I can do to improve the duration?

**Comment by xd009642** [comment]
Well, from my benchmarking it was faster than the llvm tools on parsing the files... However, rust current generates the instrumentation for all your dependencies and your crate meaning if you have some big dependencies there is a lot of data generated which just gets filtered out. There's an RFC for crate/workspace specific rustflags which would vastly improve things but until that lands the main cause (too many irrelevant traces) won't be solve-able :disappointed:

**Comment by Aeradriel** [comment]
Oh ok so it is "normal" that the run takes that much time (about 4 hours and a half vs ~15/20m for cargo test)? I indeed have big dependencies in this project (rocket, diesel, reqwest...) but I was not expecting to have that much time between the test files.
Thanks anyway for taking time to answer. I guess I'll have to wait for that RFC.
